content/projects/gsoc/2023/project-ideas/agent_reconnections_exponential_backoff.adoc

```diff
@@ -0,0 +1,59 @@
+---
+layout: gsocprojectidea
+title: "Exponential backoff and jitter for agent reconnections"
+goal: "Goal"
+category: Functional improvement
+year: 2023
+status: draft
+sig: platform
+skills:
+- Java
+mentors:
+- "basil"
+links:
+    emailThread: https://community.jenkins.io/t/gsoc-2023-project-idea-exponential-backoff-and-jitter-for-agent-reconnections/5032
+//   gitter: "jenkinsci/plugin-installation-manager-cli-tool"
+//   draft: https://docs.google.com/document/d/1s-dLUfU1OK-88bCj-GKaNuFfJQlQNLTWtacKkVMVmHc
+---
+=== Background
+
+Jenkins administrators are often facing a particular problem when restarting a large controller. It can trigger a thundering herd of agents trying to reconnect.
+Remoting and Swarm (a Remoting wrapper) have (separate) retry implementations. Swarm's supports exponential backoff but not jitter. 
+Remoting's supports neither. 
+Users have requested both in the form of bug reports and pull requests. 
+But initial pull requests need a lot of improvement before they can be merged. 
+Additionally, it would be desirable to unify the two implementations rather than to continue to maintain separate logic.
+
+
+This would be a good GSoC project for the following reasons:
+
+* Intellectually stimulating and therefore a high "brag factor": who couldn't resist spending a summer working on a topic that the link:https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/[AWS Architecture Blog] discusses?
+* High demand: Companies like Netflix needs this feature and would likely be willing to test it
+* Feasible within three months: while Remoting is a complex subsystem, retrying logic is at the outer edges of the subsystem. 
+One does not need to concern oneself with what happens inside of the guts of Remoting, just how to restart the connection process at a high level.
+* Stretchable and shrinkable scope: If the intern is struggling, scope can be reduced to unification of the existing implementations, or merely implementing exponential backoff rather than exponential backoff and jitter. 
+If the intern is doing better than expected, the scope can be expanded to writing a stress testing framework for the end result.
+* Not fatal if the project fails: Even if all the intern manages to complete is a unification of the existing two implementations, that would be a net benefit. If the project fails, no harm: the status quo would simply remain.
+
+// === Quick Start
+// TBD
+//
+// === Skills to Study and Improve
+// * TBD
+//
+=== Project Difficulty Level
+
+Intermediate
+
+=== Project Size
+
+175 hours
```

Comment:
Even the description a few lines above on line 32 ("feasible within 3 month") is incompatible with this project size. My thinking when I wrote the project description was for this to be a 350-hour project. Why was I not consulted before this PR was merged?

Reply:
@basil please review corrective PR #5963

Reply:
LGTM, thank you